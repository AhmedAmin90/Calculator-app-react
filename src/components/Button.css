.Button {
   width: 25%;
   color: black;
   text-align: center;
   border: 1px #D1D1D1 solid;
}

.Button-btn {
    width: 100%;
    height: 100%;
    border: none;
    background-color: transparent;
    color: black;
    font-size: 25px;
    font-weight: bold;
}

.orange-color {
    background-color: orange;
}

.width-double {
    width: 50%;
}