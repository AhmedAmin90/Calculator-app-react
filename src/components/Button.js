import React from 'react';
import PropTypes from 'prop-types';

const Button = ({ name, clickHandler }) => {
  const handleClick = () => {
    clickHandler(name);
  };
  return (
    <div>
      <button type="button" onClick={handleClick}>
        {' '}
        {name}
      </button>
    </div>
  );
};

Button.propTypes = {
  name: PropTypes.string,
  clickHandler: PropTypes.func.isRequired,
};

Button.defaultProps = {
  name: null,
};

export default Button;
