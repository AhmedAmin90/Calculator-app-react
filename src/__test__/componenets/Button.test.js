import React from 'react';
import { create } from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import Button from '../../components/Button';

test('Takes Name , color and clickHandler props', () => {
const elem = create(<Button color="#E0E0E0" name="AC" clickHandler={() => 'This is a test'}/>);
expect(elem).toMatchSnapshot();
  render(<Button color="#E0E0E0" name="AC" clickHandler={() => 'This is a test'} />);
  const btn = screen.getByRole('main');
  const btnName = screen.getByText('AC');
  expect(btnName).toBeString;
  expect(btnName).toBeInTheDocument;
  expect(btn.style.backgroundColor).toBe('rgb(224, 224, 224)');
});
