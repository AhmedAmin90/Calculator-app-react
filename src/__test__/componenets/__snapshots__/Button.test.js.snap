exports[`test Takes Name , color and clickHandler props 1`] = `
<div
  className="Button "
  role="main"
  style={
    Object {
      "backgroundColor": "#E0E0E0",
    }
  }>
  <button
    className="Button-btn"
    onClick={[Function]}
    type="button">
     
    AC
  </button>
</div>
`;
